src/axl.c

```diff
@@ -280,11 +335,31 @@ int AXL_Create (axl_xfer_t xtype, const char* name)
      *       SOURCE
      *     STATE
      *       CREATED */
-    kvtree* file_list = kvtree_set_kv_int(axl_file_lists, AXL_KEY_HANDLE_UID, id);
-    kvtree_util_set_str(file_list, AXL_KEY_UNAME, name);
-    kvtree_util_set_int(file_list, AXL_KEY_XFER_TYPE, xtype);
-    kvtree_util_set_int(file_list, AXL_KEY_STATUS, AXL_STATUS_SOURCE);
-    kvtree_util_set_int(file_list, AXL_KEY_STATE, (int)AXL_XFER_STATE_CREATED);
+    if (reload_from_state_file) {
+        if (axl_get_info(id, &file_list, &old_xtype, &old_xstate) != AXL_SUCCESS) {
+            AXL_ERR("Couldn't get kvtree info\n");
+            return AXL_FAILURE;
```

Comment:
Let's return -1 instead of AXL_FAILURE through all of Create for the same reason.